sdk/core/azure-core/src/http/request.cpp

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <string>
+
+#include <http/http.hpp>
+
+using namespace azure::core::http;
+using namespace std;
+
+std::string Header::getName() { return this->name; }
+std::string Header::getValue() { return this->value; }
+
+http_method::HttpMethod Request::getMethod() { return this->method; }
+string Request::getUrl() { return this->url; }
+string Request::getBody() { return this->body; }
+vector<Header> Request::getHeaders() { return this->headers; }
+
+void Request::addHeader(string name, string value)
+{
+  try
+  {
+    this->headers.push_back(Header(name, value));
+  }
+  catch (exception e)
+  {
+    throw e;
+  }
+}
+
+void Request::addQueryParameter(string name, string value)
+{
+  // Add question mark if there are not query parameters
+  if (this->query_start == 0)
+  {
+    this->url = this->url + "?";
```

Comment:
`this->url += '?'` should be enough, we don't need to make another temporary object.